packages/amplify_authenticator/lib/src/widgets/authenticator_banner.dart

```diff
@@ -21,46 +21,87 @@ import 'package:flutter/material.dart';
+
+  // If there is no background color provided by the theme,
+  // use on Colors.black87 (High-emphasis) as fallback, otherwise
+  // set it to null to let flutter control the foreground.
+  final hasCustomBackground = theme.bannerTheme.backgroundColor != null;
+  final foregroundColorFallBack = hasCustomBackground ? null : Colors.black87;
+
+  // use contentTextStyle color if available, otherwise use fallback as defined above.
+  final foregroundColor =
+      theme.bannerTheme.contentTextStyle?.color ?? foregroundColorFallBack;
 
   return MaterialBanner(
     key: keyAuthenticatorBanner,
-    leading: Icon(type.icon),
-    padding: EdgeInsetsDirectional.only(
-      start: 16.0,
-      top: 2.0 + margin,
-      bottom: 4.0,
+    leading: Icon(
+      type.icon,
+      color: foregroundColor,
     ),
-    content: content,
-    actions: actions,
+    backgroundColor: backgroundColor,
+    content: Center(
+        child: Text(
+      message.trimRight(),
+      textAlign: TextAlign.center,
+      style: TextStyle(color: foregroundColor),
+    )),
+    actions: [
+      IconButton(
+        onPressed: actionCallback,
+        icon: Icon(Icons.close, color: foregroundColor),
+      ),
+    ],
   );
 }
 
 SnackBar createSnackBar(
   BuildContext context, {
   required StatusType type,
-  required Widget content,
+  required String message,
 }) {
   final theme = Theme.of(context);
 
-  // if contentTextStyle is null, colorScheme.surface is used as a fallback
-  // since SnackBarThemeData.backgroundColor uses colorScheme.onSurface as a fall back
+  // foreground and background colors are based on material standards for text legibility
+  // if they are not provided by the theme. See https://material.io/design/color/text-legibility.html#text-backgrounds
+
+  // fallback on default color based on type.
+  final backgroundColor = theme.snackBarTheme.backgroundColor ?? type.color;
+
+  // If there is no background color provided by the theme,
+  // use on Colors.black87 (High-emphasis) as fallback, otherwise
+  // set it to null to let flutter control the foreground.
+  final hasCustomBackground = theme.snackBarTheme.backgroundColor != null;
+  final foregroundColorFallBack = hasCustomBackground ? null : Colors.black87;
+
+  // use contentTextStyle color if available, otherwise use fallback as defined above.
   final foregroundColor =
-      theme.snackBarTheme.contentTextStyle?.color ?? theme.colorScheme.surface;
+      theme.snackBarTheme.contentTextStyle?.color ?? foregroundColorFallBack;
 
   return SnackBar(
     key: keyAuthenticatorBanner,
+    backgroundColor: backgroundColor,
     content: Row(
       children: [
         Icon(
           type.icon,
           color: foregroundColor,
         ),
         const SizedBox(width: 16),
-        Expanded(child: content),
+        Expanded(
+          child: Text(message.trimRight(),
+              textAlign: TextAlign.center,
```

Comment:
The text is aligned off center here due to the icon and SizedBox on the left.

Here is the current state.
```
|-icon-|-SizedBox-|--------Content--------|
```


You could either remove the SizedBox to center it within the space not occupied by the icon, or you could center it within the full view (ignoring the icon).

However, I'm wondering if we actually want to left align this content for both SnackBar and MaterialBanner.

The Material Design specs have this content left aligned for both components. There isn't any guidance specifically NOT to center it, so I don't think we have to follow this if we have a good reason to deviate. I can see the argument that on large screens having the content in the center feels more natural, but I also think that having the content so far away from the icon is odd. My eye first jumps to the content, then to the icon, and then finally to the close button. With the icon and content left aligned my eye jumps right to the icon/content, and then to the close button.
- https://material.io/components/snackbars#specs
- https://material.io/components/banners#specs (see Mobile & Desktop sub headers)

Reply:
Resolved offline.